### PR TITLE
Fix issues with lambda labs machines

### PR DESCRIPTION
**Pull request by luke-lombardi**
- Add optional skip tls verification flag (disabled by default)
- Encapsulate remote config creation into a helper function
- Clean up pool_external and ensure gpu count is set properly on remote machines
- Pass remote config as JSON for simplicity
- Remove secret mount stuff as it's no longer needed if passing config as JSON into an env var
- Add total cpu / total mem / total gpu count fields to track workers maximum capacity

### CODE REVIEW

File: pkg/scheduler/pool_external.go [hunk 1]
```diff
@@ -119,15 +120,15 @@ func (wpc *ExternalWorkerPoolController) AddWorker(cpu int64, memory int64, gpuT
 			}
 
 			for _, worker := range workers {
-				machine.State.Cpu -= worker.Cpu
-				machine.State.Memory -= worker.Memory
-				machine.State.GpuCount -= worker.GpuCount
+				machine.State.Cpu -= worker.TotalCpu
+				machine.State.Memory -= worker.TotalMemory
+				machine.State.GpuCount -= uint32(worker.TotalGpuCount)
```
Comment (jsun-m) [inline pkg/scheduler/pool_external.go:125]: Do we need this log?

File: pkg/scheduler/pool_external.go [hunk 2]
```diff
@@ -199,10 +204,12 @@ func (wpc *ExternalWorkerPoolController) createWorkerOnMachine(workerId, machine
 		return nil, err
 	}
 
+	log.Printf("Created worked: %+v\n", worker)
+
 	return worker, nil
 }
```
Comment (jsun-m) [inline pkg/scheduler/pool_external.go:210]: Remove log

Comment (luke-lombardi) [reply pkg/scheduler/pool_external.go:208]: Done

File: pkg/scheduler/pool_external.go (hunk 1 above)
Comment (luke-lombardi) [reply pkg/scheduler/pool_external.go:125]: Removed

**Review by jsun-m — APPROVED** [review]

**Pull request merged**